### PR TITLE
Created integration test for TeacherController.

### PR DESCRIPTION
**Pull request by KrlSed**

### CODE REVIEW

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Teacher.java [hunk 1]
```diff
@@ -6,19 +6,22 @@
...
 
 /**
  * This object is entity teacher.
  *
  * @author Kirill Sedliarov
  */
 @Entity
-@Table(name = "teacher")
+@Table(name = "teachers",
+    uniqueConstraints = @UniqueConstraint(name = "UniqueFirstAndSecondName", columnNames = {"firstName", "secondName"}))
```
Comment (borecjudo2) [inline service/src/main/java/com/sedliarov/learningtable/model/entity/Teacher.java]:
```suggestion
    uniqueConstraints = @UniqueConstraint(name = "TeacherUniqueFirstAndSecondName", columnNames = {"firstName", "secondName"}))
```

File: service/src/main/java/com/sedliarov/learningtable/model/enums/MessageCode.java [hunk 2]
```diff
@@ -20,7 +20,17 @@ public enum MessageCode {
   /**
    * Error already exist linking in messages.properties value error.student.already.exist
    */
-  ERROR_STUDENT_ALREADY_EXIST("error.student.already.exist");
+  ERROR_STUDENT_ALREADY_EXIST("error.student.already.exist"),
+
+  /**
+   * Error not found linking in messages.properties value error.teacher.not.found
```
Comment (borecjudo2) [inline service/src/main/java/com/sedliarov/learningtable/model/enums/MessageCode.java]: Maybe add  a bit clarification that it for Teacher

File: service/src/test/java/com/sedliarov/learningtable/utils/TeacherFixture.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    Teacher teacher = createEntity();
+    teacher.setFirstName(firstName);
+    teacher.setSecondName(secondName);
+    return teacher;
+  }
+
+  /**
+   * Method for create Teacher entity and set admin.
+   */
+  public Teacher createEntityWithAdmin(boolean admin) {
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/utils/TeacherFixture.java]:
```suggestion
  public Teacher createEntityWithAdmin(boolean isAdmin) {
```

Comment (borecjudo2) [reply service/src/test/java/com/sedliarov/learningtable/utils/TeacherFixture.java]: and for other

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    // when
+    ResponseEntity<TeacherDto[]> response = exchangeGetWithoutAuth(TEACHERS_URL, TeacherDto[].class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).containsExactlyElementsOf(expectedTeachers);
+  }
+
+  @Test
+  void testGetTeachersIfNoUsers() {
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: Naming

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+  @Test
+  void testDeleteTeacher() {
+    // given
+    Teacher teacherForSave = TeacherFixture.createEntity();
+    teacherRepository.save(teacherForSave);
+
+    // when
+    ResponseEntity<TeacherDto> response =
+        exchangeDeleteWithoutAuth(TEACHERS_URL + teacherForSave.getTeacherId(), TeacherDto.class);
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: To be sure that we won't have lazyException
```suggestion
    Teacher teacherForSave = TeacherFixture.createEntity();
    Teacher savedTeacher = teacherRepository.save(teacherForSave);

    // when
    ResponseEntity<TeacherDto> response =
        exchangeDeleteWithoutAuth(TEACHERS_URL + savedTeacher.getTeacherId(), TeacherDto.class);
```

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    TeacherDto teacherDtoForUpdate =
+        TeacherFixture.createDtoWithFirstAndSecondName(FIRST_NAME_MARIA, SECOND_NAME_SHARAPOVA);
+
+    // when
+    ResponseEntity<TeacherDto> response =
+        exchangePutWithoutAuth(TEACHERS_URL + savedTeacher.getTeacherId(), teacherDtoForUpdate, TeacherDto.class);
+
+    // then
+    ResponseEntity<TeacherDto> checkResponse =
+        exchangeGetWithoutAuth(TEACHERS_URL + savedTeacher.getTeacherId(), TeacherDto.class);
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: We really don't need it, do we?

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+    assertSoftly(softly -> {
+      softly.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NO_CONTENT);
+      softly.assertThat(checkResponse.getBody().getTeacherId()).isEqualTo(savedTeacher.getTeacherId());
+      softly.assertThat(checkResponse.getBody().getSecondName()).isEqualTo(teacherDtoForUpdate.getSecondName());
+      softly.assertThat(checkResponse.getBody().getFirstName()).isEqualTo(teacherDtoForUpdate.getFirstName());
+      softly.assertThat(checkResponse.getBody().isAdmin()).isEqualTo(teacherDtoForUpdate.isAdmin());
+      softly.assertThat(checkResponse.getBody().getGroup()).isEqualTo(teacherDtoForUpdate.getGroup());
+    });
+  }
```
Comment (borecjudo2) [inline service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: And what about adding TeacherDto updatedTeacher = response.getBody() and use this object for assertions

File: service/src/main/java/com/sedliarov/learningtable/model/entity/Teacher.java (hunk 1 above)
Comment (KrlSed) [reply service/src/main/java/com/sedliarov/learningtable/model/entity/Teacher.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/utils/TeacherFixture.java (hunk 3 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/utils/TeacherFixture.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java (hunk 4 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java (hunk 5 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: Done

File: service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java (hunk 7 above)
Comment (KrlSed) [reply service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java]: Done

File: service/src/main/java/com/sedliarov/learningtable/model/enums/MessageCode.java (hunk 2 above)
Comment (KrlSed) [reply service/src/main/java/com/sedliarov/learningtable/model/enums/MessageCode.java]: Done

**Review by borecjudo2 — APPROVED** [review]

**Pull request merged**